File: Solutions/0228- Summary Ranges/Solution.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        answer = list()
        index = 0

        for i in range(0,len(nums)):
            if i == len(nums) - 1 or nums[i] + 1 != nums[i+1]:
                if index == i:
                    answer.append(str(nums[index]))
                else:
                    answer.append(str(nums[index])+"->"+str(nums[i]))
                index = i + 1
        
        return answer
        

File: Solutions/0228- Summary Ranges/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> answer = new ArrayList<>();
        int index = 0;

        for (int i = 0; i < nums.length; i++) {
            if (i == nums.length - 1 || nums[i] + 1 != nums[i+1]) {
                if (index == i)
                    answer.add(String.valueOf(nums[i]));
                else
                    answer.add(nums[index]+"->"+nums[i]);

                index = i + 1;
            }
        }
        return answer;
    }
}